### DISCUSSION TITLE
Invalid YouTube URL

### DISCUSSION THREAD

**Discussion by mcox00941** [danielmiessler/Fabric#760] [Q&A]
I have been successfully using the yt --transcript feature since I got Fabric working over the weekend but I'm finding some YouTube URL's throw this error. I thought it was because they didn't have a usable transcript but I visit YT and can see them there. Here is an example:

~/fabric/patterns$ yt --transcript https://www.youtube.com/live/JO71FrNxp1M
Invalid YouTube URL

I tried --comments and --metadata with the same result.

I've noticed that it seems to consistently fail on URL's with /live/ in the path. Not sure if there is something special about those or not.

**Comment by anandramya** [comment]
I get the same error. Very frustrating! I wonder the Youtube data API is what is reffered to before to add when setup. BTW what is google API? There is no documentation around this!

**Reply by mcox00941** [reply]
You have to have an account with Youtube and other services and generate an API key for Fabric to use. The video by Network Chuck in the intro touches on it. Additional details are in this discussion post:

https://github.com/danielmiessler/fabric/issues/469

**Comment by stuboo** [comment]
I think [line 16 of yt.py](https://github.com/danielmiessler/fabric/blob/d75ea473e64c387c69fb59ad021bd38f8b507ae3/installer/client/cli/yt.py#L16) needs to be updated to include the `/live/` url pattern.

ChatGPT suggests the following, but I haven't had time to test it.
```
pattern = r"(?:https?:\/\/)?(?:www\.)?(?:youtube\.com\/(?:[^\/\n\s]+\/\S+\/|(?:v|e(?:mbed)?)\/|\S*?[?&]v=|live\/)|youtu\.be\/)([a-zA-Z0-9_-]{11})"
```

**Reply by stuboo** [reply]
Okay. I just did a quick test using the URL you mentioned above and it worked.

Here's how to do it locally. I ~~may~~ submit**ted** [my first pull request](https://github.com/danielmiessler/fabric/pull/761) so that this can (maybe) go into the full repository. Here's how to make the change locally if you want though.

1. Change line 16 of yt.py as above. If you're worried, just comment out the existing line 16 and paste the line above into its place.
2. Navigate to your `fabric` directory in terminal/shell.
3. Type `pipx install . --force`
4. Type `fabric --update`
5. Restart your terminal/shell

**Reply by mcox00941** [reply]
Nice work! I'll give that a try when I can.

**Reply by mcox00941** [reply]
Sorry if this is a dumb question but which one do I edit?

find ~/ -type f -name "yt.py"
/home/user/fabric/installer/client/cli/yt.py
/home/user/.local/pipx/venvs/fabric/lib/python3.10/site-packages/installer/client/cli/yt.py

**Reply by stuboo** [reply]
Hard to say. It looks to me like you tried to install it twice (once in a virtual environement). It shouldn't break anything either way.  If you're concerned, simply comment out the line that starts with `pattern` by putting a `#` in front of `pattern` then you can go back and undo your changes if something breaks.

**Comment by matigumma** [comment]
i think live events can not be transcribed as same way of  a normal video.

**Reply by stuboo** [reply]
I think that's true _if_ the event is streaming, but after the stream is over and it's archived, the live url still works and a transcription can become available. For example, the link above was a live event and now a transcript is available via the youtube api.

**Comment by prometheusone** [comment]
or use the right youtube url , just righclick the video and copy the url as its formatted there

**Reply by stuboo** [reply]
Works like a charm. Thank you.

**Reply by mcox00941** [reply]
Ah yes. I kept trying the 'share' url which still included /live but I see if you right click it's different. Thank you!
